src/components/Dashboard/index.tsx

```diff
@@ -173,6 +173,12 @@ export default class Dashboard extends React.Component<IDashboardProps, IDashboa
   }
 
   onDeleteDashboardApprove() {
+    let { dashboard } = this.props;
+    if (!dashboard) {
+      console.warn('Dashboard not found. Aborting delete.');
```

Comment:
Please show a Toast instead (so that the user can see) and then call `return;` so it wouldn't try to actually delete the dashboard.